app/views/collection_numbers/_form.html.erb

```diff
@@ -1,29 +1,26 @@
-<% # form_tag(add_query_param(action: :new, id: @observation.id)) do %>
-<%= form_with model: :collection_number do |f| %>
+<%= form_with model: @collection_number do |f| %>
 
-  <%= f.submit button.t, class: "btn mx-auto mt-3" %>
+  <%= submit_tag(local_assigns[:button].t, class: "btn mx-auto mt-3") %>
```

Comment:
The combination of locals: { button: :SAVE } in the view and button in the form was not working (for reasons I don't understand.

Reply:
Yes, i've run into that too, but usually with partials-within-partials. Something about the form partial is now called by the new and edit templates, the locals have to be accessed via local_assigns[:attr]. However i would change this to f.submit local_assigns[:button].t, class="btn mx-auto mt-3" so rails can do whatever it wants to do with ID's by binding the submit button to the whole form instance "f"

Reply:
To give a concrete example i believe this should be
`f.submit local_assigns[:button].t, class: "btn mx-auto mt-3"`
The "f" binds it to the form, and changes the syntax - you don't need the "_tag" at that point